### PR TITLE
Store baselines locally on streak2.

### PR DESCRIPTION
**Pull request by CusiniM**
Introduces the use a persistent folder on streak2 to store baselines locally instead of downloading them.

The folder is:

`/data/GEOS/baselines`

and baselines are stored as:

`/data/GEOS/baselines/baselines_pr#pr/baseline_integratedTests-pr-#pr-${{ github.run_number }}-${SHORT_COMMIT}.tar.gz`

### CODE REVIEW

**Comment by CusiniM** [comment]
It's now failing coz I had made a mistake and created a file that can't be deleted but this should work once someone deletes that with sudo. 

It just needs a bit more testing.

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -241,20 +264,32 @@ jobs:
...
+              echo "Baselines ${DATA_EXCHANGE_DIR}/baseline_${DATA_BASENAME} were not uploaded. Likeyly because no rebaseline was necessary."
+            fi
           fi
         fi
         
         # manually remove the workspace to avoid issues with the next job when using self-hosted runners
         if [ -d "${GITHUB_WORKSPACE}/integratedTests" ]; then
+          # Matteo: IMO this is useless now.
           rm -rf ${GITHUB_WORKSPACE}/integratedTests
         fi
```
Comment (CusiniM) [inline .github/workflows/build_and_test.yml]:
```suggestion
```

Comment (CusiniM) [reply .github/workflows/build_and_test.yml]: I don't think this is useful?

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -241,20 +277,32 @@ jobs:
...
-                echo "New baseline ID: baseline_${DATA_BASENAME::-7}"
-              else
-                echo "Baselines ${DATA_EXCHANGE_DIR}/baseline_${DATA_BASENAME} were not uploaded. Likeyly because no rebaseline was necessary."
-              fi
-            # fi
+            if [ -f ${DATA_EXCHANGE_DIR}/baselines/baseline_${DATA_BASENAME} ];then
+              # 1. We copy the baselines to a local directory to store them 
+                
+              # 1.a Create the new target directory to store the new baselines
+              TARGET_DIR="${{ inputs.LOCAL_BASELINE_DIR }}/baseline_integratedTests-pr-${{ github.event.number }}"
```
Comment (CusiniM) [inline .github/workflows/build_and_test.yml]:
```suggestion
              TARGET_DIR="${{ inputs.LOCAL_BASELINE_DIR }}/pr-${{ github.event.number }}"
```

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -205,6 +205,42 @@ jobs:
...
+          echo "Baseline: ${BASELINE_TAG}"
+
+          # Extract the folder name
+          BASELINE_FOLDER_NAME=$(echo "$BASELINE_TAG" | grep -o 'pr[0-9]*')
+          echo "Baseline folder name: ${BASELINE_FOLDER_NAME}"
+
+          CURRENT_BASELINE_DIR=${{ inputs.LOCAL_BASELINE_DIR }}/${BASELINE_FOLDER_NAME}
+          mkdir -p ${CURRENT_BASELINE_DIR}
+
+          wget -P ${CURRENT_BASELINE_DIR} https://storage.googleapis.com/geosx/integratedTests/baseline_integratedTests-pr3125-5101-7764ffb.tar.gz
```
Comment (CusiniM) [inline .github/workflows/build_and_test.yml]:
```suggestion
```

Comment (cssherman) [inline .github/workflows/build_and_test.yml]: Looks like this should be unnecessary now?

File: .github/workflows/build_and_test.yml [hunk 4]
```diff
@@ -205,6 +205,42 @@ jobs:
...
+          # Remove the 'integratedTests/' prefix
+          BASELINE_TAG=${BASELINE_FULL_PATH#integratedTests/}
+          echo "Baseline: ${BASELINE_TAG}"
+
+          # Extract the folder name
+          BASELINE_FOLDER_NAME=$(echo "$BASELINE_TAG" | grep -o 'pr[0-9]*')
+          echo "Baseline folder name: ${BASELINE_FOLDER_NAME}"
+
+          CURRENT_BASELINE_DIR=${{ inputs.LOCAL_BASELINE_DIR }}/${BASELINE_FOLDER_NAME}
+          mkdir -p ${CURRENT_BASELINE_DIR}
```
Comment (CusiniM) [inline .github/workflows/build_and_test.yml]:
```suggestion
```

**Review by cssherman — APPROVED** [review]

File: .github/workflows/build_and_test.yml [hunk 5]
```diff
@@ -205,6 +205,46 @@ jobs:
...
+          # Remove the 'integratedTests/' prefix
+          BASELINE_TAG=${BASELINE_FULL_PATH#integratedTests/}
+          echo "Baseline: ${BASELINE_TAG}"
+
+          # Extract the folder name
+          PR_NUMBER=$(echo "$BASELINE_TAG" | grep -o 'pr[0-9]*')
+          PR_BASELINE_FOLDER_NAME=baselines_${PR_NUMBER}
+          echo "Baseline folder name: ${BASELINE_FOLDER_NAME}"
+
+          CURRENT_BASELINE_DIR=${{ inputs.LOCAL_BASELINE_DIR }}/${BASELINE_FOLDER_NAME}
```
Comment (CusiniM) [inline .github/workflows/build_and_test.yml]:
```suggestion
          CURRENT_BASELINE_DIR=${{ inputs.LOCAL_BASELINE_DIR }}/${PR_BASELINE_FOLDER_NAME}
```

File: .github/workflows/build_and_test.yml [hunk 6]
```diff
@@ -205,6 +205,46 @@ jobs:
...
+          BASELINE_FULL_PATH=$(grep -A 2 'baselines:' "${YAML_FILE_PATH}" | grep 'baseline:' | awk '{print $2}')
+
+          # Remove the 'integratedTests/' prefix
+          BASELINE_TAG=${BASELINE_FULL_PATH#integratedTests/}
+          echo "Baseline: ${BASELINE_TAG}"
+
+          # Extract the folder name
+          PR_NUMBER=$(echo "$BASELINE_TAG" | grep -o 'pr[0-9]*')
+          PR_BASELINE_FOLDER_NAME=baselines_${PR_NUMBER}
+          echo "Baseline folder name: ${BASELINE_FOLDER_NAME}"
```
Comment (CusiniM) [inline .github/workflows/build_and_test.yml]:
```suggestion
          echo "Baseline folder name: ${PR_BASELINE_FOLDER_NAME}"
```

File: src/CMakeLists.txt [hunk 7]
```diff
@@ -232,8 +232,8 @@ install( FILES ${CMAKE_BINARY_DIR}/schema.xsd
 ################################
 # Add python environment setup
 ################################
-# message(WARNING "Temporarily changing the geosPythonBranch to feature/sherman/baselineStorage")
-# set(GEOS_PYTHON_PACKAGES_BRANCH "branch name" CACHE STRING "" FORCE)
+message(WARNING "Temporarily changing the geosPythonBranch to cusini/fix-streak-cert-fail")
+set(GEOS_PYTHON_PACKAGES_BRANCH "cusini/fix-streak-cert-fail" CACHE STRING "" FORCE)
```
Comment (CusiniM) [inline src/CMakeLists.txt]:
```suggestion
# message(WARNING "Temporarily changing the geosPythonBranch to cusini/fix-streak-cert-fail")
# set(GEOS_PYTHON_PACKAGES_BRANCH "cusini/fix-streak-cert-fail" CACHE STRING "" FORCE)
```

**Pull request merged**